module/solutions/helper.js

```diff
@@ -694,4 +743,280 @@ module.exports = class SolutionsHelper {
+     * @param {String} templateId - solution template id.
+     * @param {Object} program
+     * @param {String} program.id - program id
+     * @param {String} program.name - program name
+     * @param {String} userId - Logged in user id.
+     * @param {Object} solutionData - new solution creation data
+     * @param {Boolean} [isAPrivateProgram = false] - created program is private or not     
+     * @returns {Object} Created solution and program
+     */
+
+    static createProgramAndSolutionFromTemplate(
+      templateId,
+      program,
+      userId,
+      solutionData,
+      isAPrivateProgram = false
+  ) {
+      return new Promise(async (resolve, reject) => {
+          try {
+
+              let dateFormat = gen.utils.epochTime();
+              let programData;
+
+              if( program.id === "" ) {
+
+                programData = await programsHelper.create({
+                  externalId : program.name + "-" + dateFormat ,
+                  description : solutionData.description,
+                  name : program.name,
+                  userId : userId,
+                  isAPrivateProgram : isAPrivateProgram
+                });
+                
+                program.id = programData._id;
+              }
+
+              let solutionQuery =   {
+                _id : templateId
+              };
+
+              let programQuery = {
+                _id : program.id
+              };
+
+              let duplicateSolution = 
+              await this.importFromSolution(
+                solutionQuery,
+                programQuery,
+                userId,
+                solutionData,
+                false
+              );
+
+              return resolve(
+                _.pick(
+                  duplicateSolution,
+                  [
+                    "_id",
+                    "externalId",
+                    "frameworkExternalId",
+                    "frameworkId",
+                    "programExternalId",
+                    "programId",
+                    "entityTypeId",
+                    "entityType",
+                    "isAPrivateProgram"
+                  ]
+                  ));
+
+          } catch (error) {
+              return reject(error);
+          }
+      })
+  }
+
+      /**
+     * Import solution from template.
+     * @method
+     * @name importFromSolution -
+     * @param {Object} solutionQuery - solution query data
+     * @param {Object} programQuery - program query data.
+     * @param {String} userId - logged in user id.
+     * @param {Object} data - new solution data  
+     * @returns {Object} New solution information
+     */
+
+    static importFromSolution(
+      solutionQuery,
+      programQuery,
+      userId,
+      data,
+      isReusable = true
+    ) {
+      return new Promise(async (resolve, reject) => {
+        try {
+
+          let solutionDocument = await this.solutionDocuments(
+            solutionQuery
```

Comment:
@aman-tunerlabs - Why to expect a query, can it not be just solutionExternalId

Reply:
@aks30 Front end is sending solution id in create api so instead of making database call to fetch solution external id. I made it to work for solutionExternalId and solutionId .

Reply:
@aman-tunerlabs - So yeah just pass the id, why pass a solutionQuery as anobject